test/util/test_app.go

```diff
@@ -73,23 +76,99 @@ func SetupTestAppWithGenesisValSet(cparams *tmproto.ConsensusParams, genAccounts
 	return testApp, kr
 }
 
-func NewTestAppWithGenesisSet(cparams *tmproto.ConsensusParams, genAccounts ...string) (*app.App, *tmtypes.ValidatorSet, keyring.Keyring) {
-	// var cache sdk.MultiStorePersistentCache
+// NewTestApp creates a new app instance with an empty memDB and a no-op logger.
+func NewTestApp() *app.App {
 	// EmptyAppOptions is a stub implementing AppOptions
 	emptyOpts := EmptyAppOptions{}
 	// var anteOpt = func(bapp *baseapp.BaseApp) { bapp.SetAnteHandler(nil) }
 	db := dbm.NewMemDB()
 
 	encCfg := encoding.MakeConfig(app.ModuleEncodingRegisters...)
 
-	testApp := app.New(
+	return app.New(
 		log.NewNopLogger(), db, nil,
 		cast.ToUint(emptyOpts.Get(server.FlagInvCheckPeriod)),
 		encCfg,
 		0,
 		emptyOpts,
 	)
+}
+
+// ApplyGenesisState sets genesis on initialized testApp with the provided arguments.
+func ApplyGenesisState(testApp *app.App, pubKeys []cryptotypes.PubKey, balance int64, cparams *tmproto.ConsensusParams) (keyring.Keyring, []genesis.Account, error) {
```

Comment:
why not?
```suggestion
func ApplyGenesisState(testApp *app.App, gen *genesis.Genesis) error {
```
The way you currently have it is not applying genesis but creating a genesis.

Reply:
preferably i'd have a function that lets you customise genesis without having to do the setup yourself. i'm happy to change the function signature to correctly reflect the behaviour. we also have [a ticket refactoring test_app ](https://github.com/celestiaorg/celestia-app/issues/2535) so we'll spend more time properly designing this process.

Reply:
technically this method is applying genesis on already initialised testapp